### ISSUE TITLE
Solving http.request.method cardinality (and similar) with metrics hints API

### ISSUE THREAD

**Issue by lmolkova** [open-telemetry/semantic-conventions#68]
Original issue: https://github.com/open-telemetry/opentelemetry-specification/issues/3470 

TL;DR: `http.request.method` can have high cardinality, especially when a malicious or buggy client sends requests with invalid methods. Servers don't have control over this behavior.

The problem is generic and could be solved with hint API that
- allows applications to specify known/prioritized values to record
- SDK follows hint API and records known values, and records unknown ones with 'unknown'/'other' (TBD) value

In the context of `http.request.method`:
- general-purpose instrumentation libraries SHOULD configure a hint with standard method names
- SDK users MAY override it and allow additional methods

This approach is outlined in #35 

**Pros:**
- generic solution not specific to one attribute

**Cons:**
- cannot be applied to other signals if they need to have low cardinality (e.g. span names) 
- can't query across signals for attributes with capped values.

**Comment by lmolkova** [comment]
This can now be closed as we went ahead with approach in #17
